networkx/algorithms/assortativity/tests/test_mixing.py

```diff
@@ -33,9 +33,8 @@ def test_degree_mixing_dict_multigraph(self):
 class TestDegreeMixingMatrix(BaseTestDegreeMixing):
     def test_degree_mixing_matrix_undirected(self):
         # fmt: off
-        a_result = np.array([[0, 0, 0],
-                             [0, 0, 2],
-                             [0, 2, 2]]
+        a_result = np.array([[0, 2],
+                             [2, 2]]
```

Comment:
The mixing matrix is a square matrix where the number of rows (columns) is equal to the number of unique values in the data. Here is the degree mixing matrix of a path graph. Path graph has only two unique degrees: 1 and 2. Therefore, the mixing matrix should be 2x2. The values are correct: 2 links from degree 1 to 2, 2 links from degree 2 to 1, 2 links from 2 to 2 and 0 links from degree from 1 to 1 (in directed representation).

Reply:
This represents a change in behavior for `degree_mixing_matrix` --- it sounds like maybe the original definition was incorrect? Either way, I think we should at least detail the behavior change in a release note.